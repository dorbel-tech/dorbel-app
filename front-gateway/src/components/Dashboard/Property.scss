@import "../sharedStyles/variables.scss";
@import "../sharedStyles/bsvars.scss";
@import "../sharedStyles/tab-menu.scss";
@import "./style/property.scss";
@import "./style/propertyMenu.scss";
