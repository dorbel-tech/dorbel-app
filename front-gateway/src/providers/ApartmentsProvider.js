/**
 * ApartmentsProvider communicates with the Apartments API
 */
'use strict';
import { action } from 'mobx';
import _ from 'lodash';

class ApartmentsProvider {
  constructor(appStore, providers) {
    this.appStore = appStore;
    this.apiProvider = providers.api;
    this.cloudinaryProvider = providers.cloudinary;
    this.oheProvider = providers.ohe;
  }

  loadApartments(query) {
    const q = encodeURIComponent(JSON.stringify(query || {}));
    return this.apiProvider.fetch('/api/apartments/v1/listings?q=' + q)
      .then(this.appStore.listingStore.add);
  }

  loadFullListingDetails(id) {
    return Promise.all([
      this.apiProvider.fetch('/api/apartments/v1/listings/' + id)
        .then(action('load-single-apartment', apartment => this.appStore.listingStore.listingsById.set(id,apartment))),
      this.oheProvider.loadListingEvents(id),
      this.oheProvider.getFollowsForListing(id)
    ]);
  }

  mapUploadApartmentFormToCreateListing(formValues) {
    let listing = {};
    // this is so we can use nested structure in our form attributes
    Object.keys(formValues).filter(key => formValues.hasOwnProperty(key)).forEach(key => _.set(listing, key, formValues[key]));
    listing.images = formValues.images.map((cloudinaryImage, index) => ({
      url: cloudinaryImage.secure_url, display_order: index
    }));

    return listing;
  }

  uploadApartment(formValues) {
    const listing = this.mapUploadApartmentFormToCreateListing(formValues);
    return this.apiProvider.fetch('/api/apartments/v1/listings', { method: 'POST', data: listing })
      .then(newListing => this.oheProvider.createOhe(Object.assign({ listing_id: newListing.id }, formValues.open_house_event)));
  }

  @action
  uploadImage(file) {
    const imageStore = this.appStore.newListingStore.formValues.images;
    const image = { complete: false, src: file.preview, progress: 0 };
    imageStore.push(image);

    const onProgress = action('image-upload-progress', e => image.progress = e.lengthComputable ? (e.loaded / e.total) : 0);

    return this.cloudinaryProvider.upload(file, onProgress)
    .then(action('image-upload-done', uploadedImage => {
      image.complete = true;
      image.src = `http://res.cloudinary.com/dorbel/${uploadedImage.resource_type}/${uploadedImage.type}/v${uploadedImage.version}/${uploadedImage.public_id}.${uploadedImage.format}`;
      image.delete_token = uploadedImage.delete_token;
      image.secure_url = uploadedImage.secure_url;
      return uploadedImage;
    }))
    .catch(action(() => {
      imageStore.remove(image); // remove method is available as this is a mobx observable array
    }));
  }

  @action
  deleteImage(image) {
    this.appStore.newListingStore.formValues.images.remove(image);
    return this.cloudinaryProvider.deleteImage(image);
  }

  updateListingStatus(listingId, status) {
    return this.apiProvider.fetch('/api/apartments/v1/listings/' + listingId, { method: 'PATCH', data: { status } })
    .then((res) => {
      let listing = this.appStore.listingStore.listingsById.get(listingId);
      listing.status = status;
      _.set(listing, 'meta.possibleStatuses', _.get(res, 'meta.possibleStatuses'));
    });
  }

  getRelatedListings(listingId){
    return this.apiProvider.fetch('/api/apartments/v1/listings/'+listingId+'/related/');
  }
}

module.exports = ApartmentsProvider;
