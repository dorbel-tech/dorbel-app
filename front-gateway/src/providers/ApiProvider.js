/**
 * ApiProvider supplies infrastructure for calling the API layer
 */
'use strict';
import axios from 'axios';
let urlPrefix = '';

if (!process.env.IS_CLIENT) {
  // TODO: on the server we should make direct calls to the API's
  // instead of that we are making the calls from the server to itself and they go through the gateway
  urlPrefix = 'http://localhost:' + require('dorbel-shared').config.get('PORT');
}

class ApiProvider {
  constructor(appStore) {
    this.appStore = appStore;
  }

  fetch(url, options){
    const headers = {
      'Accept': 'application/json',
      'Content-Type': 'application/json'
    };

    if (this.appStore.authStore.isLoggedIn) {
      headers['Authorization'] = 'Bearer ' + this.appStore.authStore.getToken();
    }

    return axios({
      url: urlPrefix + url,
      headers,
      ...options
    })
    .then(res => res.data);
  }
}

module.exports = ApiProvider;

