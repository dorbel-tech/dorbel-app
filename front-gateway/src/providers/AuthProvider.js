'use strict';
import promisify from 'es6-promisify';

class AuthProvider {
  constructor(clientId, domain, authStore, router) {
    const Auth0Lock = require('auth0-lock').default; // can only be required on client side
    this.lock = new Auth0Lock(clientId, domain, {
      auth: {
        redirectUrl: window.location.origin + '/login',
        responseType: 'token'
      },
      initialScreen: 'signUp',
      theme: {
        logo: 'https://res.cloudinary.com/dorbel/image/upload/c_scale,h_58,w_58/v1477485453/dorbel_logo_2_1_uvvf3j.png',
        primaryColor: '#31124b'
      },
      languageDictionary: {
        title: 'הרשמו לזיהוי קל בהמשך ולעדכונים חשובים בלבד',
        welcome: 'היי %s!',
        signUpWithLabel: '%s',
        signupTitle: 'הרשמה',
        signUpLabel: 'הרשמה',
        signUpSubmitLabel: 'הרשמה',
        loginLabel: 'כניסה',
        loginSubmitLabel: 'כניסה',
        loginWithLabel: '%s',
        databaseAlternativeSignUpInstructions: 'או',
        databaseEnterpriseAlternativeLoginInstructions: 'או',
        forgotPasswordAction: 'שכחתם סיסמא?',
        forgotPasswordInstructions: 'הכניסו את כתובת המייל שלכם. אנו נשלח אליכם מייל לאיפוס סיסמא.',
        forgotPasswordSubmitLabel: 'שלח',
        signUpTerms: ' אני מסכים/ה <a href="https://www.dorbel.com/pages/terms" target="_blank">לתנאי השימוש </a><a href="https://www.dorbel.com/pages/privacy_policy" target="_blank">ומדיניות הפרטיות</a>',
        signUp: {
          user_exists: 'כתובת מייל קיימת, לחצו על ״כניסה״ ונסו שנית',
        },
        login: {
          'lock.invalid_email_password': 'מייל זה לא קיים, לחצו על כפתור הרשמה והרשמו לאתר',
        },
        success: {
          forgotPassword: 'מייל לאיפוס ססמתך נשלח אלייך זה עתה',
        }
      },
      mustAcceptTerms: true
    });
    this.lock.on('authenticated', this.afterAuthentication.bind(this));
    this.authStore = authStore;
    this.router = router;
    this.showLoginModal = this.showLoginModal.bind(this);
    this.logout = this.logout.bind(this);
    this.reportIdentifyAnalytics(this.authStore.profile);
  }

  afterAuthentication(authResult) {
    this.authStore.setToken(authResult.idToken);
    this.getProfile(authResult)
    .then(() => { // wait until profile is set because our previous state might depend on it
      if (authResult.state) {
        this.recoverStateAfterLogin(authResult.state);
      }
    });
  }

  recoverStateAfterLogin(stateString) {
    try {
      const stateBeforeLogin = JSON.parse(stateString);
      if (stateBeforeLogin && stateBeforeLogin.pathname) {
        this.router.setRoute(stateBeforeLogin.pathname);
      }
    } catch(ex) {
      window.console.error('error parsing state after login');
    }
  }

  getProfile(authResult) {
    // DEPRECATION NOTICE: This method will be soon deprecated, use `getUserInfo` instead
    return promisify(this.lock.getProfile, this.lock)(authResult.idToken)
    .then(profile => {
      let mappedProfile = this.mapAuth0Profile(profile);
      this.authStore.setProfile(mappedProfile);
      this.reportIdentifyAnalytics(mappedProfile);
    })
    .catch(error => {
      window.console.log('Error loading the Profile', error);
      throw error;
    });
  }

  mapAuth0Profile(auth0profile) {
    const user_metadata = auth0profile.user_metadata || {};
    const app_metadata = auth0profile.app_metadata || {};

    return Object.assign({}, auth0profile, {
      first_name: user_metadata.first_name || auth0profile.given_name,
      last_name: user_metadata.last_name || auth0profile.family_name,
      email: user_metadata.email || auth0profile.email,
      phone: user_metadata.phone || auth0profile.phone,
      role: app_metadata.role || 'user'
    });
  }

  showLoginModal() {
    this.lock.show({
      auth: {
        params: {
          state: JSON.stringify({
            pathname: window.location.pathname
          })
        }
      }
    });
  }

  logout() {
    this.authStore.logout();
  }

  reportIdentifyAnalytics(profile) {
    // https://segment.com/docs/integrations/intercom/#identify
    if (profile) { window.analytics.identify(profile.dorbel_user_id, profile); }
  }
}

module.exports = AuthProvider;
