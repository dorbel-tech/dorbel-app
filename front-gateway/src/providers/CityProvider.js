/**
 * City provider communicates with the Apartments API to get city meta data
 */
'use strict';

const getCitiesQuery = `
  query GetCities {
    cities {
      id
      city_name
      display_order
    }
  }
`;

class CityProvider {
  constructor(appStore, apiProvider) {
    this.appStore = appStore;
    this.apiProvider = apiProvider;
    this.isLoadingCities = false;
  }

  loadCities() {
    return this.apiProvider.gql(getCitiesQuery).then(({ data }) => this.appStore.cityStore.cities = data.cities);
  }
}

module.exports = CityProvider;
