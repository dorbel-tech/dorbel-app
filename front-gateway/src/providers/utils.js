/***
 * General utils modeule
 */
'use strict';
import _ from 'lodash';
import moment from 'moment';

const timeFormat = 'HH:mm';
const dateFormat = 'DD/MM/YY';
const dayFormat = 'dd';

function formatTime(date) {
  return moment.utc(date).local().format(timeFormat);
}

function formatDate(date) {
  return moment.utc(date).local().format(dateFormat);
}

function formatDay(date) {
  return moment.utc(date).local().format(dayFormat);
}

function getListingTitle(listing) {
  return listing.title || `דירת ${listing.apartment.rooms} חד׳ ב${listing.apartment.building.street_name}`;  
}

function sortListingImages(listing) {
  return listing.images.length ? _.orderBy(listing.images, ['display_order']) : [];
}

function getListingStatusLabels() {
  return {
    pending: { label: 'ממתינה לאישור', actionLabel: 'החזר את הדירה להמתנה' },
    listed: { label: 'מפורסמת', actionLabel: 'פרסם את הדירה' },
    rented: { label: 'מושכרת', actionLabel: 'הדירה הושכרה' },
    unlisted: { label: 'לא פעילה', actionLabel: 'השהה מודעה' },
    deleted: { label: 'נמחקה', actionLabel: 'מחק מודעה' }
  };
}


module.exports = {
  formatTime,
  formatDate,
  formatDay,
  getListingTitle,
  sortListingImages,
  getListingStatusLabels
};
