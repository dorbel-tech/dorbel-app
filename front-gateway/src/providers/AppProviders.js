/**
 * AppProviders will consolidate all the providers in the application
 * Providers should supply data to the stores, but not store any data themselves
 */
'use strict';
import AuthProvider from './AuthProvider';
import ApiProvider from './ApiProvider';
import ListingsProvider from './ListingsProvider';
import OheProvider from './OheProvider';
import CityProvider from './CityProvider';
import NeighborhoodProvider from './NeighborhoodProvider';
import CloudinaryProvider from './CloudinaryProvider';

const isServer = !global.window;

class AppProviders {
  constructor(appStore, router) {
    if (!isServer) {
      if (!window.dorbelConfig.AUTH0_FRONT_CLIENT_ID || !window.dorbelConfig.AUTH0_DOMAIN) {
        throw new Error('must set auth0 env vars');
      }
      this.authProvider = new AuthProvider(window.dorbelConfig.AUTH0_FRONT_CLIENT_ID, window.dorbelConfig.AUTH0_DOMAIN, appStore.authStore, router);
    }

    this.cloudinaryProvider = new CloudinaryProvider();

    this.apiProvider = new ApiProvider(appStore);
    this.oheProvider = new OheProvider(appStore, this.apiProvider);
    this.apartmentsProvider = new ApartmentsProvider(appStore, 
      { api: this.apiProvider, cloudinary: this.cloudinaryProvider, ohe: this.oheProvider });
    this.cityProvider = new CityProvider(appStore, this.apiProvider);
    this.neighborhoodProvider = new NeighborhoodProvider(appStore, this.apiProvider);
  }
}

module.exports = AppProviders;
