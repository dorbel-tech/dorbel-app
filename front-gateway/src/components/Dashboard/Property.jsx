import React, { Component } from 'react';
import { inject, observer } from 'mobx-react';
import autobind from 'react-autobind';
import { Button, Col, Grid, Row, OverlayTrigger, Popover } from 'react-bootstrap';
import LoadingSpinner from '~/components/LoadingSpinner/LoadingSpinner';
import CloudinaryImage from '../CloudinaryImage/CloudinaryImage';
import ListingStatusSelector from './MyProperties/ListingStatusSelector';
import OHEManager from '~/components/OHEManager/OHEManager';
import PropertyMenu from './MyProperties/PropertyMenu';
import PropertyStats from './MyProperties/PropertyStats';
import { find } from 'lodash';
import utils from '~/providers/utils';

import './Property.scss';

@inject('appStore', 'appProviders', 'router') @observer
class Property extends Component {
  constructor(props) {
    super(props);
    autobind(this);

    this.state = { isLoading: false };
  }

  static serverPreRender(props) {
    return props.appProviders.listingsProvider.loadFullListingDetails(props.propertyId);
  }

  componentWillReceiveProps(nextProps) {
    if (this.props.propertyId != nextProps.propertyId) {
      this.props = nextProps;
      this.loadFullPropertyDetails();
    }
  }

  componentWillMount() {
    this.loadFullPropertyDetails();
  }

  loadFullPropertyDetails() {
    let propertyId = this.props.propertyId;
    if (!this.props.appStore.listingStore.get(propertyId)) {
      this.setState({ isLoading: true });
      this.props.appProviders.listingsProvider.loadFullListingDetails(propertyId)
        .then(() => this.setState({ isLoading: false }));
    }
  }

  routeClickHandler(e) {
    this.props.router.setRoute(e.target.getAttribute('name'));
  }

  refresh() {
    location.reload(true);
  }

  render() {
    const { appStore } = this.props;
    const property = appStore.listingStore.get(this.props.propertyId);
    const sortedPropertyImages = utils.sortListingImages(property);
    const imageURL = sortedPropertyImages.length ? sortedPropertyImages[0].url : '';
    const followers = appStore.oheStore.countFollowersByListingId.get(this.props.propertyId);

    const propertyTabs = [
      { relativeRoute: 'stats', title: 'סטטיסטיקות', component: <PropertyStats listing={property} followers={followers || 0} /> },
      { relativeRoute: 'ohe', title: 'מועדי ביקור', component: <OHEManager listing={property} /> },
    ];
    // TODO: Add "default" tab logic.
    const selectedTab = find(propertyTabs, {relativeRoute: this.props.tab}) || propertyTabs[0];

    const popoverMenu = (
      <Popover id="property-actions-menu" className="property-actions-menu">
        <div name={'/apartments/' + this.props.propertyId} className="property-actions-menu-item property-action-menu-item-show-mobile" onClick={this.routeClickHandler}>
          <i className="property-actions-menu-item-icon fa fa-picture-o"></i>
          צפה
        </div>
        <div className="property-actions-menu-item property-action-menu-item-show-mobile" onClick={this.refresh}>
          <i className="property-actions-menu-item-icon fa fa-refresh" aria-hidden="true"></i>
          רענון
        </div>
        <div name={'/dashboard/my-properties/' + this.props.propertyId + '/edit'}
             className="property-actions-menu-item property-actions-edit-disabled"
             {/*onClick={this.routeClickHandler}*/}>
          <i className="property-actions-menu-item-icon fa fa-pencil-square-o"  aria-hidden="true"></i>
          עריכת פרטי הנכס (בקרוב)
        </div>
      </Popover>
    );

    if (this.state.isLoading) {
      return (
        <div className="loader-container">
          <LoadingSpinner />
        </div>
      );
    }

    return  <Grid fluid className="property-wrapper">
              <Row className="property-top-container">
                <Col md={3} sm={3} xs={5} className="property-image-container">
                  <CloudinaryImage src={imageURL} height={125} className="property-image"/>
                  <ListingStatusSelector listing={property} />
                </Col>
                <Col md={5} sm={6} xs={7} className="property-title-container">
                  <div className="property-title">
                    {utils.getListingTitle(property)}
                  </div>
                  <div className="property-title-details">
                    <span>
                      {property.apartment.rooms}</span>
                      <span className="property-title-details-sub-text"> חדרים</span>
                    <div className="property-title-details-vr" />
                    <span>
                      {property.apartment.size}</span>
                      <span className="property-title-details-sub-text"> מ"ר</span>
                    <div className="property-title-details-vr" />
                    <span>
                      <span className="property-title-details-sub-text property-title-details-last-text">קומה</span>
                      {utils.getFloorTextValue(property)}</span>
                  </div>
                </Col>
                <Col md={4} sm={3} className="property-actions-wrapper">
                  <div className="property-action-container property-actions-details">
                    <div>
                      <span className="property-actions-title">
                        {followers || 0}</span><br/>
                        <span className="property-actions-sub-title">עוקבים</span>
                    </div>
                    <div className="property-actions-vr" />
                    <div>
                      <span className="property-actions-title">
                        {property.totalLikes || 0}</span><br/>
                        <span className="property-actions-sub-title">לייקים</span>
                    </div>
                  </div>
                  <div className="property-action-container">
                    <div className="property-actions-refresh-container">
                      <Button className="fa fa-refresh property-action-button" aria-hidden="true"
                         onClick={this.refresh}></Button>
                    </div>
                    <div className="property-actions-preview-container">
                      <Button className="fa fa-picture-o property-action-button"
                              name={'/apartments/' + this.props.propertyId}
                              onClick={this.routeClickHandler}></Button>
                    </div>
                    <div className="property-actions-menu-container">
                      <OverlayTrigger trigger="click" placement="bottom" overlay={popoverMenu}
                                      container={this} containerPadding={5} rootClose>
                        <Button className="property-action-button">
                          <i className="fa fa-bars" aria-hidden="true"></i>
                        </Button>
                      </OverlayTrigger>
                    </div>
                  </div>
                </Col>
              </Row>
              <PropertyMenu path={'/dashboard/my-properties/' + property.id + '/'}
                            tabs={propertyTabs}
                            activeKey={selectedTab.relativeRoute} />
              <Row className="property-content-container">
                {selectedTab.component}
              </Row>
            </Grid>;
  }
}

Property.wrappedComponent.propTypes = {
  propertyId: React.PropTypes.string.isRequired,
  tab: React.PropTypes.string,
  appProviders: React.PropTypes.object,
  appStore: React.PropTypes.object,
  router: React.PropTypes.object
};

export default Property;
