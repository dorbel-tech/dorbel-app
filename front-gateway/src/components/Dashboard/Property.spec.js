import React from 'react';
import { shallow, mount } from 'enzyme';
import { Provider } from 'mobx-react';

import Property from './Property';
import { flushPromises } from '~/providers/utils';

describe('Property', () => {
  let appStoreMock, appProvidersMock, propertyMock;

  beforeEach(() => {
    // because we mount Property (not just shallow), we need to mock all the dependencies of its children
    appStoreMock = {
      listingStore: {
        get: jest.fn(),
        listingViewsById: { get: jest.fn(), has: jest.fn() },
        isListingPublisherOrAdmin: jest.fn().mockReturnValue(true),
        listingsByApartmentId: { get: jest.fn() }
      },
      oheStore: {
        oheByListingId: jest.fn()
      },
      editedListingStore: {
        loadListing: jest.fn()
      },
      likeStore: {
        likesByListingId: { 
          get: jest.fn().mockReturnValue([]), 
          has: jest.fn() 
        }
      }
    };
    appProvidersMock = {
      listingsProvider: {
        loadListingPageViews: jest.fn(),
        loadFullListingDetails: jest.fn(),
        loadListingsForApartment: jest.fn(),
        isActiveListing: jest.fn().mockReturnValue(true)
      },
      oheProvider: {
        loadListingEvents: jest.fn()
      },
      navProvider: {},
      likeProvider: {
        getLikesForListing: jest.fn()
      }
    };
    propertyMock = {
      id: 7,
      apartment_id: 8,
      status: 'rented',
      apartment: {
        building: {}
      },
      images: [ { url: 'bla' } ]
    };
  });

  const shallowProperty = () => shallow(<Property.wrappedComponent appStore={appStoreMock} appProviders={appProvidersMock} listingId={'' + propertyMock.id} />);
  // TODO : mount is !!! terrible !!! here it needs a million dependencies - find a way to test without mount
  const mountProperty = () => mount(
    <Provider appStore={appStoreMock} appProviders={appProvidersMock} router={{}}>
      <Property listingId={'' + propertyMock.id} />
    </Provider>
  );

  it('should render in loading state', () => {
    const wrapper = shallowProperty();
    expect(wrapper.find('LoadingSpinner')).toHaveLength(1);
    expect(wrapper.state('isLoading')).toBe(true);
    expect(wrapper).toMatchSnapshot();
  });

  it('should not be in loading state when the store has the property', () => {
    appStoreMock.listingStore.get.mockReturnValue(propertyMock);
    const wrapper = mountProperty();

    // this depends on ComponentDidMount + rerender after state.isLoading was changed
    return flushPromises().then(() => {
      expect(appStoreMock.editedListingStore.loadListing).toHaveBeenCalledWith(propertyMock);
      wrapper.update();
      expect(wrapper.find('LoadingSpinner')).toHaveLength(0);
    });
  });

  it('should start in loading state and then change when the provider gets the listing', () => {
    appProvidersMock.listingsProvider.loadFullListingDetails.mockReturnValue(Promise.resolve());
    const wrapper = mountProperty();

    expect(wrapper.find('LoadingSpinner')).toHaveLength(1);
    expect(appProvidersMock.listingsProvider.loadFullListingDetails).toHaveBeenCalledWith('' + propertyMock.id);
    appStoreMock.listingStore.get.mockReturnValue(propertyMock);

    // this depends on ComponentDidMount + rerender after state.isLoading was changed
    return flushPromises().then(() => {
      expect(appStoreMock.editedListingStore.loadListing).toHaveBeenCalledWith(propertyMock);
      wrapper.update();
      expect(wrapper.find('LoadingSpinner')).toHaveLength(0);
    });
  });
});
