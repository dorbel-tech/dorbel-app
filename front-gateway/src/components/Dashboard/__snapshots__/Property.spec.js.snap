exports[`Property should render in loading state 1`] = `
<div
  className="loader-container">
  <LoadingSpinner />
</div>
`;
